internal/handlers/sqlite/msg_find.go

```diff
@@ -99,11 +99,14 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 	}
 
 	// Limit pushdown is not applied if:
-	//  - `filter` is set, it must fetch all documents to filter them in memory;
-	//  - `sort` is set but `EnableSortPushdown` is not set, it must fetch all documents
-	//  and sort them in memory;
+	//  - `filter` is set but can not be applied pushdown entirely,
+	//  it must fetch all documents to filter them in memory;
+	//  - `sort` is set but can not be applied pushdown entirely,
+	//  or `sort` is set but `EnableSortPushdown` is not set,
+	//  it must fetch all documents and sort them in memory;
 	//  - `skip` is non-zero value, skip pushdown is not supported yet.
-	if params.Filter.Len() == 0 && (params.Sort.Len() == 0 || h.EnableSortPushdown) && params.Skip == 0 {
+	// More checks are performed when preparing LIMIT clause.
+	if (params.Sort.Len() == 0 || h.EnableSortPushdown) && params.Skip == 0 {
```

Comment:
We cannot set limit pushdown if `h.DisableFilterPushdown` is true and filter is set. That's because filter is expected to be done purely in memory, so we need all documents fetched without any `LIMIT`. :hugs:

Reply:
handler code is updated to take `h.DisableFilterPushdown` into consideration.